registry_module_integration_test.go

```diff
@@ -1323,6 +1324,60 @@ func TestRegistryModulesRead(t *testing.T) {
 	})
 }
 
+func TestRegistryModulesReadRegistry(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+	r := require.New(t)
+
+	// create an org that will be deleted later. the wskp will live here
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	org, err := client.Organizations.Read(ctx, orgTest.Name)
+	r.NoError(err)
+	r.NotNil(org)
+
+	githubIdentifier := os.Getenv("GITHUB_REGISTRY_NO_CODE_MODULE_IDENTIFIER")
+	if githubIdentifier == "" {
+		t.Skip("Export a valid GITHUB_REGISTRY_NO_CODE_MODULE_IDENTIFIER before running this test")
+	}
+
+	token, cleanupToken := createOAuthToken(t, client, org)
+	defer cleanupToken()
+
+	rmOpts := RegistryModuleCreateWithVCSConnectionOptions{
+		VCSRepo: &RegistryModuleVCSRepoOptions{
+			OrganizationName:  String(org.Name),
+			Identifier:        String(githubIdentifier),
+			Tags:              Bool(true),
+			OAuthTokenID:      String(token.ID),
+			DisplayIdentifier: String(githubIdentifier),
+		},
+	}
+
+	version := "1.0.0"
+	rm, err := client.RegistryModules.CreateWithVCSConnection(ctx, rmOpts)
+	r.NoError(err)
+
+	// Wait a few seconds to let the module become ready
+	time.Sleep(time.Second * 5)
+
+	t.Run("fetch module from registry", func(t *testing.T) {
+		rmID := RegistryModuleID{
+			Organization: orgTest.Name,
+			Name:         rm.Name,
+			Provider:     rm.Provider,
+			Namespace:    rm.Namespace,
+			RegistryName: rm.RegistryName,
+		}
+		tfm, err := client.RegistryModules.ReadRegistry(ctx, rmID, version)
+		r.NoError(err)
+		r.NotNil(tfm)
+		r.Equal(fmt.Sprintf("%s/%s/%s/%s", orgTest.Name, rm.Name, rm.Provider, version), tfm.ID)
+		r.Equal(version, tfm.Version)
```

Comment:
what do you think about adding checks for expected `TerraformRegistryModule` attributes in addition to the Version attribute?

Reply:
I added several more checks. I did not update the test for every attribute of the module, but many of them. Also, several of the test values are hard-coded, making this test very brittle to changes in the [module under test](https://github.com/hashicorp/terraform-random-module). For example, if a new version of that module is published with one additional input variable, then the `Len` check for the `Inputs` field will fail.

Just wanted to call that out! 😄 I still think it's good that other attributes are being verified.